### PR TITLE
Parametrize khiops version

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: docs/setup/pip.md [hunk 1]
```diff
@@ -13,25 +13,25 @@ We support :simple-python: **Python from 3.8 to 3.12**.
     sudo apt-get update -y && sudo apt-get install wget lsb-release -y && \
     CODENAME=$(lsb_release -cs) && \
     TEMP_DEB="$(mktemp)" && \
-    wget -O "$TEMP_DEB" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi_10.2.3-1-${CODENAME}.amd64.deb" && \
+    wget -O "$TEMP_DEB" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi_{{ KHIOPS_VERSION }}-1-${CODENAME}.amd64.deb" && \
     sudo dpkg -i "$TEMP_DEB" || sudo apt-get -f -y install && \
     rm -f $TEMP_DEB && \
-    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/10.2.3.0/khiops-10.2.3.0.tar.gz'
+    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/{{ KHIOPS_PYTHON_VERSION }}/khiops-{{ KHIOPS_PYTHON_VERSION }}.tar.gz'
```
Comment (popescu-v) [inline docs/setup/pip.md:19]: We need to remove "-" from `KHIOPS_PYTHON_VERSION` in `khiops-{{ KHIOPS_PYTHON_VERSION }}.tar.gz`.

File: docs/setup/pip.md [hunk 2]
```diff
@@ -13,25 +13,25 @@ We support :simple-python: **Python from 3.8 to 3.12**.
...
+    <a href="https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-{{ KHIOPS_VERSION }}-setup.exe">
         <button class="btn btn-light btn-sm">
           Download for Windows
         </button>
     </a>
 
     Then, you can run the following Pip command:
     ```sh
-    pip install "https://github.com/KhiopsML/khiops-python/releases/download/10.2.3.0/khiops-10.2.3.0.tar.gz"
+    pip install "https://github.com/KhiopsML/khiops-python/releases/download/{{ KHIOPS_PYTHON_VERSION }}/khiops-{{ KHIOPS_PYTHON_VERSION }}.tar.gz"
```
Comment (popescu-v) [inline docs/setup/pip.md:34]: We need to remove "-" from release version in the tarball (see same comment above).

File: docs/setup/pip.md [hunk 3]
```diff
@@ -42,10 +42,10 @@ We support :simple-python: **Python from 3.8 to 3.12**.
     sudo yum update -y && sudo yum install wget python3-pip -y && \
     CENTOS_VERSION=$(rpm -E %{rhel}) && \
     TEMP_RPM="$(mktemp).rpm" && \
-    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
     sudo yum install "$TEMP_RPM" -y && \
     rm -f $TEMP_RPM && \
-    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/10.2.3.0/khiops-10.2.3.0.tar.gz'
+    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/{{ KHIOPS_PYTHON_VERSION }}/khiops-{{ KHIOPS_PYTHON_VERSION }}.tar.gz'
```
Comment (popescu-v) [inline docs/setup/pip.md:48]: Remove "-" from release version in the tarball (see above).

File: mkdocs.yml [hunk 4]
```diff
@@ -97,9 +102,10 @@ extra_javascript:
 
 plugins:
   - mkdocs-jupyter
-  #  - blog
+  - macros
```
Comment (popescu-v) [inline mkdocs.yml:105]: Why this addition? To support the Jinja2 templating mechanism?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+This is the technical repository for the Khiops documentation website.
+
+## Setup
+Install the website requirements
+
+```bash
+pip install -r requirements.txt
+```
+
+Then, to run locally just execute
```
Comment (popescu-v) [inline README.md:11]: Add mention that we need to set-up the environment variables for the release versions of khiops, khiops python, viz, coviz, in order for the templates to get correctly instantiated?

**Review by popescu-v — CHANGES_REQUESTED** [review]
A few comments.

File: README.md (hunk 5 above)
Comment (folmos-at-orange) [reply README.md:11]: That is not part of the setup.

File: mkdocs.yml (hunk 4 above)
Comment (folmos-at-orange) [reply mkdocs.yml:105]: yes

File: docs/setup/pip.md [hunk 6]
```diff
@@ -13,25 +16,25 @@ We support :simple-python: **Python from 3.8 to 3.12**.
     sudo apt-get update -y && sudo apt-get install wget lsb-release -y && \
     CODENAME=$(lsb_release -cs) && \
     TEMP_DEB="$(mktemp)" && \
-    wget -O "$TEMP_DEB" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi_10.2.3-1-${CODENAME}.amd64.deb" && \
+    wget -O "$TEMP_DEB" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi_{{ KHIOPS_VERSION }}-1-${CODENAME}.amd64.deb" && \
     sudo dpkg -i "$TEMP_DEB" || sudo apt-get -f -y install && \
     rm -f $TEMP_DEB && \
-    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/10.2.3.0/khiops-10.2.3.0.tar.gz'
+    pip install 'https://github.com/KhiopsML/khiops-python/releases/download/{{ TARGZ_KHIOPS_PYTHON_VERSION }}/khiops-{{ KHIOPS_PYTHON_VERSION }}.tar.gz'
```
Comment (popescu-v) [inline docs/setup/pip.md:22]: It's the other way round: in the release URL (_except_ the tar.gz), we need to keep the `KHIOPS_PYTHON_VERSION`; the `TARGZ_KHIOPS_PYTHON_VERSION` needs to be put in the tar.gz file.

Whence:

```sh
pip install 'https://github.com/KhiopsML/khiops-python/releases/download/{{ KHIOPS_PYTHON_VERSION }}/khiops-{{ TARGZ_KHIOPS_PYTHON_VERSION }}.tar.gz'
```

File: docs/setup/nocode.md [hunk 7]
```diff
@@ -67,8 +67,8 @@ For further details, you may refer to [README.txt][readme], and [whatsnew.txt][w
     CENTOS_VERSION=$(rpm -E %{rhel}) && \
     TEMP_RPM="$(mktemp).rpm" && \
     TEMP_RPM_KHIOPS="$(mktemp).rpm" && \
-    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
-    wget -O "$TEMP_RPM_KHIOPS" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM_KHIOPS" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
```
Comment (popescu-v) [inline docs/setup/nocode.md:71]: Replace "-" with "_" in pre-release field (see e.g. https://github.com/KhiopsML/khiops/releases/download/10.2.3-rc.2/khiops-core-openmpi-10.2.3_rc.2-1.el8.x86_64.rpm)

Comment (folmos-at-orange) [reply docs/setup/nocode.md:71]: Ok

**Review by popescu-v — CHANGES_REQUESTED** [review]

File: docs/setup/visualization.md [hunk 8]
```diff
@@ -12,31 +12,31 @@ You can find all versions of the Khiops Visualization tool on the [the following
...
+- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v{{ KHIOPS_VIZ_VERSION }}/khiops-visualization-{{ KHIOPS_VIZ_VERSION }}.dmg">
           **macOS x86-64**  </a>  (Intel-based)
-- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v11.0.9/khiops-visualization-11.0.9-arm64.dmg">
+- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v{{ KHIOPS_VIZ_VERSION }}/khiops-visualization-{{ KHIOPS_VIZ_VERSION }}-arm64.dmg">
           **macOS ARM** </a>   (:material-apple: Apple Silicon - M1/M2/M3)
-- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v11.0.9/khiops-visualization_11.0.9_amd64.deb">
+- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v{{ KHIOPS_VIZ_VERSION }}/khiops-visualization_{{ KHIOPS_VIZ_VERSION }}_amd64.deb">
           **Ubuntu** </a>
-- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v11.0.9/khiops-visualization-11.0.9.x86_64.rpm">
+- <a href="https://github.com/KhiopsML/kv-electron/releases/download/v{{ KHIOPS_VIZ_VERSION }}/khiops-visualization-{{ KHIOPS_VIZ_VERSION }}.x86_64.rpm">
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply docs/setup/visualization.md:23]: Done

File: docs/setup/pip.md [hunk 9]
```diff
@@ -42,10 +45,10 @@ We support :simple-python: **Python from 3.8 to 3.12**.
     sudo yum update -y && sudo yum install wget python3-pip -y && \
     CENTOS_VERSION=$(rpm -E %{rhel}) && \
     TEMP_RPM="$(mktemp).rpm" && \
-    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply docs/setup/pip.md]: Done

File: docs/setup/visualization.md [hunk 10]
```diff
@@ -12,31 +12,31 @@ You can find all versions of the Khiops Visualization tool on the [the following
...
+- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v{{ KHIOPS_COVIZ_VERSION }}/khiops-covisualization-{{ KHIOPS_COVIZ_VERSION }}.dmg">
           **macOS x86-64**  </a>  (Intel-based)  
-- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v11.1.2/khiops-covisualization-11.1.2-arm64.dmg">
+- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v{{ KHIOPS_COVIZ_VERSION }}/khiops-covisualization-{{ KHIOPS_COVIZ_VERSION }}-arm64.dmg">
           **macOS ARM** </a>   (:material-apple: Apple Silicon - M1/M2/M3) 
-- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v11.1.2/khiops-covisualization_11.1.2_amd64.deb">
+- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v{{ KHIOPS_COVIZ_VERSION }}/khiops-covisualization_{{ KHIOPS_COVIZ_VERSION }}_amd64.deb">
           **Ubuntu** </a>
-- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v11.1.2/khiops-covisualization-11.1.2.x86_64.rpm">
+- <a href="https://github.com/KhiopsML/kc-electron/releases/download/v{{ KHIOPS_COVIZ_VERSION }}/khiops-covisualization-{{ KHIOPS_COVIZ_VERSION }}.x86_64.rpm">
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply docs/setup/visualization.md:39]: Done

File: docs/setup/index.md [hunk 11]
```diff
@@ -111,13 +112,13 @@ This version contains a standalone Graphical User Interface (GUI).
     CENTOS_VERSION=$(rpm -E %{rhel}) && \
     TEMP_RPM="$(mktemp).rpm" && \
     TEMP_RPM_KHIOPS="$(mktemp).rpm" && \
-    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-core-openmpi-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
-    wget -O "$TEMP_RPM_KHIOPS" "https://github.com/KhiopsML/khiops/releases/download/10.2.3/khiops-10.2.3-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-core-openmpi-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
+    wget -O "$TEMP_RPM_KHIOPS" "https://github.com/KhiopsML/khiops/releases/download/{{ KHIOPS_VERSION }}/khiops-{{ KHIOPS_VERSION }}-1.el${CENTOS_VERSION}.x86_64.rpm" && \
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply docs/setup/index.md]: Done.

File: docs/setup/visualization.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (popescu-v) [reply docs/setup/visualization.md:23]: Right, we keep it as such for now.

File: docs/setup/pip.md [hunk 12]
```diff
@@ -1,3 +1,8 @@
+<!-- Transform a pre-release version to the .tar.gz file version
+{% set TARGZ_KHIOPS_PYTHON_VERSION = KHIOPS_PYTHON_VERSION.replace("-rc.", "rc").replace("-b.", "b").replace("-a.", "a") %}
+{% set ROCKY_KHIOPS_VERSION = KHIOPS_VERSION.replace("-", "_") %}
```
Comment (popescu-v) [inline docs/setup/pip.md:3]: Add mention of Rocky version in the comment as well.

File: docs/setup/nocode.md [hunk 13]
```diff
@@ -1,3 +1,6 @@
+<!-- Transform a pre-release version to the .tar.gz file version
```
Comment (popescu-v) [inline docs/setup/nocode.md:1]: Update comment to Rocky version (not tar.gz).

File: docs/setup/index.md [hunk 14]
```diff
@@ -1,3 +1,6 @@
+<!-- Transform a pre-release version to the .tar.gz file version
```
Comment (popescu-v) [inline docs/setup/index.md:1]: Update comment to Rocky version (not tar.gz).

**Review by popescu-v — APPROVED** [review]
LGTM

**Pull request merged**